src/inc/genXplatEtw.pl

```diff
@@ -0,0 +1,782 @@
+                        {
+                            my $varname = $params->{"var"} || die "no name ";
+                            if ($varname  eq $firstdependecy ){  last; }
+                            $i++;
+                        }
+
+                        my $dependecieslen =@$dependecies;
+                        splice @lttngFnFrame, $i,$dependecieslen ;
+                    }
+
+                    @{$allLTTngTemplateTypes{$lttngtemplateName}} =@lttngFnFrame;
+                    push (@{$allTemplatemap{$templateName}}, $lttngtemplateName);
+                    $allTemplateVarProps{$lttngtemplateName} =\%varProps;
+                    my @varnames =keys %lttngFnFrameShadow;
+                    #clear lttngFnFrameShadow
+                    undef %lttngFnFrameShadow;
+                    undef @lttngFnFrame ;
+                    $lttngtemplateName ="$templateName\_$subevent_cnt";
+                    $subevent_cnt++;
+                    $count=0;
+
+                    next; # so that next frame can begin
+                }
+
+                my $dependecies = $varDependencies{$dataName} || die "no varDependencies for $dataName";
+                foreach my $dependency (@$dependecies)
+                {
+
+                    if (not exists $lttngFnFrameShadow{$dependency})
+                    {
+                        $lttngFnFrameShadow{$dependency} = 1;
+                        $lttngFnFrame[$count++] = $fnSignatureshadow{$dependency};
+
+                        #   if ($dataName ne   $dependency){
+                        goto FRAMEARGS;
+                        #}
+                    }
+                }
+            }
+            my $templatelen=  @fnSignature;
+            if($templatelen==0 || $count>0)
+            {
+                push (@{$allTemplatemap{$templateName}}, $lttngtemplateName);
+                $allLTTngTemplateTypes{$lttngtemplateName} =\@lttngFnFrame;
+                $allTemplateVarProps{$lttngtemplateName} =\%varProps;
+                my @varnames =keys %lttngFnFrameShadow;
+            }
+        }
+
+
+#we loop over the template property so that the code generated is same every time the scipt runs
+#generate the code in
+
+#            generate $lttngevent*header.h
+#            each template corresponds to a TRACEPOINT_EVENT_CLASS, its structure would be like
+#define ThreadCreated_TRACEPOINT_ARGS \
+#    TP_ARGS(\
+#        uint64_t, ManagedThreadID,\
+#        uint64_t, AppDomainID,\
+#        const unsigned int, Flags,\
+#        const unsigned int, ManagedThreadIndex,\
+#        const unsigned int, OSThreadID,\
+#        const unsigned short, ClrInstanceID\
+#    )
+#TRACEPOINT_EVENT_CLASS(
+#    Microsoft_DotNETRuntime,
+#    ThreadCreated ,
+#    ThreadCreated_TRACEPOINT_ARGS,
+#    TP_FIELDS(
+#        ctf_integer(uint64_t, ManagedThreadID,ManagedThreadID)
+#        ctf_integer(uint64_t, AppDomainID,AppDomainID)
+#        ctf_integer(const unsigned int, Flags,Flags)
```

Comment:
No, it does not make sense to use "const" at all within the arguments of ctf_integer() and other similar ctf_*() macros, because this translates into an lvalue being assigned to. This is not a function input argument.